nautobot/utilities/utils.py

```diff
@@ -689,3 +704,220 @@ def pretty_str(self, node=None, depth=0):
+        form_field_class = forms.ChoiceField
+        form_field_class.widget = StaticSelect2Multiple()
+        form_attr = {"choices": field.extra.get("choices")}
+
+        form_field = form_field_class(**form_attr)
+    elif isinstance(field, (BooleanFilter,)):  # Yes / No choice
+        form_field_class = forms.ChoiceField
+        form_field_class.widget = StaticSelect2()
+        form_attr = {"choices": BOOLEAN_WITH_BLANK_CHOICES}
+
+        form_field = form_field_class(**form_attr)
+    elif isinstance(field, DateTimeFilter):
+        form_field.widget = DateTimePicker()
+    elif isinstance(field, DateFilter):
+        form_field.widget = DatePicker()
+    elif isinstance(field, TimeFilter):
+        form_field.widget = TimePicker()
+
+    form_field.required = False
+    form_field.initial = None
+    form_field.widget.attrs.pop("required", None)
+
+    css_classes = form_field.widget.attrs.get("class", "")
+    form_field.widget.attrs["class"] = "form-control " + css_classes
+    return form_field
+
+
+def get_values_display_names(model, search_by, values):
+    """
+    Get all display names for values in `model` instance.
+
+    Examples:
+        >>> get_values_display_names(,,<class: nautobot.dcim.models.Site>, "slug", ["site-1", "site-2"] )
+        >>> [("site-1", "Site 1"), ("site-2", "Site 2")]
+    Args:
+        model: Model to query
+        search_by: Filter query by
+        values: List of values to get their display names
+    """
+    choices = []
+    for value in values:
+        try:
+            instance = model.objects.get(**{search_by: value})
+            choices.append((value, getattr(instance, "display", str(instance))))
+        except model.DoesNotExist:
+            pass
+    return choices
+
+
+def convert_querydict_to_factory_formset_acceptable_querydict(request_querydict, filterset_class):
+    """
+    Convert request QueryDict/GET into an acceptable factory formset QueryDict
+    while discarding `querdict` params which are not part of `filterset_class` params
+
+    Args:
+        request_querydict (QueryDict): QueryDict to convert
+        filterset_class: Filterset class
+
+    Examples:
+        >>> convert_querydict_to_factory_formset_acceptable_querydict({"status": ["active", "decommissioning"], "name__ic": ["site"]},)
+        >>> {
+        ...     'form-TOTAL_FORMS': [3],
+        ...     'form-INITIAL_FORMS': ['0'],
+        ...     'form-MIN_NUM_FORMS': [''],
+        ...     'form-MAX_NUM_FORMS': [''],
+        ...     'form-0-lookup_field': ['status'],
+        ...     'form-0-lookup_type': ['status'],
+        ...     'form-0-value': ['active', 'decommissioning'],
+        ...     'form-1-lookup_field': ['name'],
+        ...     'form-1-lookup_type': ['name__ic'],
+        ...     'form-1-value': ['site']
+        ... }
+    """
+    query_dict = QueryDict(mutable=True)
+    filterset_class_fields = filterset_class.base_filters.keys()
```

Comment:
Just checking - does this feature work correctly for dynamically generated filters (filtering on custom fields, relationships)?